src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -1460,30 +1459,41 @@ private ITypeName GenericTypeArgumentsRule(Token genericTypeName, Token firstTok
                 }
                 else
                 {
-                    ReportIncompleteInput(After(commaOrRBracketToken),
+                    ReportIncompleteInput(
+                        After(lastToken),
                         nameof(ParserStrings.MissingTypename),
                         ParserStrings.MissingTypename);
-                    typeName = new TypeName(commaOrRBracketToken.Extent, ":ErrorTypeName:");
+                    typeName = new TypeName(lastToken.Extent, ":ErrorTypeName:");
                 }
 
                 genericArguments.Add(typeName);
             }
 
-            if (commaOrRBracketToken.Kind != TokenKind.RBracket)
+            return genericArguments;
+        }
+
+        private ITypeName GenericTypeArgumentsRule(Token genericTypeName, Token firstToken, bool unBracketedGenericArg)
```

Comment:
Given that we're breaking these methods out into separate pieces, we should try to rename them to refer to exactly what they parse. I'm not clear on the difference between `GenericArgumentsRule` and `GenericTypeArgumentsRule`

Reply:
The first should probably be `GenericMethodArgumentsRule`, but I'll double check.

Reply:
Ah, I see where the confusion comes from. Okay... the current `GenericTypeArgumentsRule` returns a `GenericTypeName`, so I'm going to rename:

- `GenericTypeArgumentsRule` => `GenericTypeNameRule`
- `GenericArgumentsRule` => `GenericTypeArgumentsRule`

That seems to make the most sense to me.

Reply:
> GenericArgumentsRule => GenericTypeArgumentsRule

This can mislead if anybody looks a commit history. Could we use another name for new `GenericTypeArgumentsRule`?

Reply:
I'm open to suggestions if the names can be clearer, but the naming of the original method this is built out of puts us in a little bit of a corner in that way.